### ISSUE TITLE
Button click event is processed before Editor data is sent to server.

### ISSUE THREAD

**Issue by TatuLund** [vaadin/vaadin-grid-pro-flow#85]
Ways to reproduce

Create Grid Pro with two or more rows in it (does not happen with only one row)

Create a "Save" Button

Add click listener to Button and display values from data provider.

Edit items in Grid Editor
Click "Save" Button

Observed behavior:
- Updated values are displayed, except for last edited item old value is displayed

Expeted behavior
- All edited values are upto date

Workaround, programmically close the editor with JavaScript and process data in call back:

```
		button.addClickListener(event -> {
			 grid.getElement().executeJs("this._stopEdit(false,true);").then(i -> {
				ListDataProvider<Person> dp = (ListDataProvider<Person>) grid.getDataProvider();			
				dp.getItems().stream().map(item -> item.getName()).forEach(name -> Notification.show(name));			
				
			});
		});


```

**Comment by TatuLund** [comment]
Complete code example can be found here:

https://github.com/TatuLund/devday-demo-flow/blob/master/src/main/java/com/vaadin/devday/demo/views/GridProView.java

**Comment by alvarezguille** [comment]
This could be related to the 200ms timeout in editor focus out debouncer in vaadin-grid-pro-inline-editing-mixin

https://github.com/vaadin/vaadin-grid-pro/blob/v2.0.6/src/vaadin-grid-pro-inline-editing-mixin.html#L245

**Comment by Haprog** [comment]
@alvarezguille You're right. I created a V15 project with a simplified GridProView similar to the linked example above. I can reproduce the issue but if I locally edit the implementation of `vaadin-grid-pro-inline-editing-mixin.html` `_onEditorFocusOut` in `node_modules` to remove the debouncing then it works even without the above `executeJs` workaround.

**Comment by Haprog** [comment]
Removing the debouncing from there breaks quite many tests in `vaadin-grid-pro` though so I tested different debouncing delays.

The current implementation uses `Polymer.Async.timeOut.after(200)`.

Here's some others I tested (on my work laptop running project locally):
- `Polymer.Async.timeOut.after(X)`
  - X = 110-200: no difference to current implementation
  - X = 70-100: Save button without workaround **works some times** but not always and all web component **tests pass**
  - X = 60: Save button without workaround **seems to work** "always" (tested at least 10 times) and all web component **tests pass**
  - X = 0: Save button without workaround **seems to work always** and all web component **tests pass**
- `Polymer.Async.microTask`
  - Save button without workaround **seems to work always** and all web component **tests pass**
- `Polymer.Async.animationFrame`
  - Save button without workaround seems to work always and all web component **tests pass**

**Comment by Haprog** [comment]
This should be fixed by https://github.com/vaadin/vaadin-grid-pro/pull/111

**Comment by Haprog** [comment]
The fix was released in `vaadin-grid-pro` [v2.0.7](https://github.com/vaadin/vaadin-grid-pro/releases/tag/v2.0.7) and `vaadin-grid-pro-flow` [2.0.4](https://github.com/vaadin/vaadin-grid-pro-flow/releases/tag/2.0.4) and [3.0.1](https://github.com/vaadin/vaadin-grid-pro-flow/releases/tag/3.0.1).
